### PR TITLE
Correlations dialog

### PR DESCRIPTION
**Pull request by shadrackkibet**

### CODE REVIEW

**Comment by shadrackkibet** [comment]
@rdstern In correlations dialog when i check correlations plot or pairwise plot radio buttons should i still be able to select a method in the main dialog?. I think the group of methods on the main dialog should be disabled in this two cases.
image

**Comment by dannyparsons** [comment]
The method relates to the `cor` function of the dialog. So if a graph is chosen then this should use the same method as selected on the dialog. At the moment it's not so this is was a good issue to find.
I think it is now possible to specify the method for `ggpairs` function https://github.com/ggobi/ggally/issues/60. Also need to check this for the correlation plot.

**Comment by shadrackkibet** [comment]
@africanmathsinitiative/developers  this is ready for review.

**Comment by shadrackkibet** [comment]
I have also enabled scatter matrix option in the subdialog and fixed bugs.So this can be tested.

image

**Comment by shadrackkibet** [comment]
@africanmathsinitiative/developers ?

**Comment by shadrackkibet** [comment]
@dannyparsons this is ready for you.

**Comment by dannyparsons** [comment]
@africanmathsinitiative/developers can someone review this?

**Comment by shadrackkibet** [comment]
@africanmathsinitiative/developers ?

**Comment by Patowhiz** [comment]
@shadrackkibet which issue is this pull request trying to solve

**Comment by shadrackkibet** [comment]
> @shadrackkibet which issue is this pull request trying to solve

There is no open issue for this. If you follow up the discussion in this pullrequest it should be clear the issue it fixes.

**Comment by Patowhiz** [comment]
This works fine on my side.
@dannyparsons  this is ready for you.

**Comment by shadrackkibet** [comment]
@dannyparsons  this is ready for you to review.

File: instat/dlgCorrelation.vb [hunk 1]
```diff
@@ -170,12 +182,17 @@ Public Class dlgCorrelation
...
+        ucrPnlMethod.AddAdditionalCodeParameterPair(clsRGGcorrGraphicsFunction, New RParameter("method", 2), iAdditionalPairNo:=3)
+        ucrPnlMethod.AddAdditionalCodeParameterPair(clsRGGscatMatrixFunction, New RParameter("corMethod", 4), iAdditionalPairNo:=4)
         '        ucrReceiverMultipleColumns.AddAdditionalCodeParameterPair(clsRGraphicsFuction, New RParameter("columns", 1), iAdditionalPairNo:=1) ' this has to be done manually for now as it needs to be a string for this function but r function for another
         ucrReceiverMultipleColumns.SetRCode(clsCorrelationFunction, bReset)
         ucrNudConfidenceInterval.SetRCode(clsCorrelationTestFunction, bReset)
         ucrReceiverFirstColumn.SetRCode(clsCorrelationTestFunction, bReset)
         ucrReceiverSecondColumn.SetRCode(clsCorrelationTestFunction, bReset)
-        ucrPnlColumns.SetRCode(ucrBase.clsRsyntax.clsBaseFunction, bReset)
+        If bReset Then
+            ucrPnlColumns.SetRCode(clsCorrelationTestFunction, bReset)
```
Comment (dannyparsons) [inline instat/dlgCorrelation.vb]: Does this not work on reopening?

**Comment by dannyparsons** [comment]
This is a great improvement to get the correlation method the same for the output and graph and also getting the scatter matrix plot working, its very nice!
One bug I noticed is that when I did ggpairs the command was:
`GGally::ggpairs(data=survey, columns=c("Field","Size","Fert"), upper=list(continuous=GGally::wrap('cor', method="spearman")), lower=list(continuous='cor'))`
So the method is only set on upper instead of lower meaning the correlation values are different on lower and upper diagonal. So the method needs to be the same on upper and lower.

Also I don't think the alpha option for ggscatmat and ggcorr are needed. I assume any of the layer parameters could be passed through here and alpha is doing the transparency of points/tiles. I think it's not particularly important to have as an option so could be removed.

File: instat/dlgCorrelation.vb (hunk 1 above)
Comment (shadrackkibet) [reply instat/dlgCorrelation.vb]: Yep this is not working on reopen. I am not sure exactly why.

**Comment by shadrackkibet** [comment]
@dannyparsons i have sorted out the bug. This is ready for re-review.

**Comment by shadrackkibet** [comment]
@dannyparsons ?

**Comment by dannyparsons** [comment]
Great, this could be very useful for our workshop next week.

**Pull request merged**